lib/src/op_heads_store.rs

```diff
@@ -32,6 +33,18 @@ pub enum OpHeadResolutionError {
     NoHeads,
 }
 
+#[derive(Debug, Error)]
+#[error(transparent)]
+pub struct OpHeadStoreError(pub Box<dyn std::error::Error + Send + Sync>);
+
+pub type OpHeadStoreResult<T> = Result<T, OpHeadStoreError>;
+
+impl From<PathError> for OpHeadStoreError {
+    fn from(err: PathError) -> Self {
+        OpHeadStoreError(err.into())
+    }
+}
```

Comment:
I wouldn't add this conversion impl.

`PathError` is as generic as `io::Error`, and implementor of `OpHeadStore` might want to map I/O-related errors to more descriptive types. `From<PathError>` can be footgun there.

Reply:
The reason I wanted to use PathError was because the path error message seemed useful and relevant in this case. Would it make sense to add the path to OpHeadStoreError then?

Reply:
The use of `PathError` is fine. My point is conversion of `path_error?` to `OpHeadStoreError` can hide future bugs.

For example, another `OpHeadsStore` backend can define its own error type containing `CustomOpHeadStoreError::Read(PathError)|Write(PathError)`, and `From<CustomOpHeadStoreError> for OpHeadStoreError`. We should use `path_err.map_err(..::Read)?`, etc. instead of `path_err?` there, but the latter would also compile if we had `From<PathError>`.